### PR TITLE
feat/display-subflow-content

### LINKED ISSUE #774: Enhance UI representation of subflows from a flow topology
### Feature description

Use-case: when dealing with several subflows or a few number of subflows with a lot of tasks, it's quite complicated to follow the execution or understand the topology when looking at the parent flow.

Suggested solutions:
- set-up a breadcrumb in order to "go back" to the calling flow, when we dived into the topology of a subflow
- create a "expand" button on a subflow node when looking at the topology of a flow in order to unpack it

Thanks :)

### PR DESCRIPTION
**Pull request by brian-mulier-p**
closes #774

### CODE REVIEW

**Comment by brian-mulier-p** [comment]
[Screencast from 31-08-2023 21:09:11.webm](https://github.com/kestra-io/kestra/assets/37618489/89e711fe-4575-4596-9b81-c9aea7bf5b5c)

**Comment by brian-mulier-p** [comment]
For the flowable part maybe I can duplicate it without the subflow part to ensure backport compat with a default for the parent subflow one but I think people will need to agree and implement this part if they want proper subflow visualization so idk

File: core/src/main/java/io/kestra/core/services/AdvancedGraphService.java [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+import jakarta.inject.Singleton;
+import lombok.extern.slf4j.Slf4j;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Stream;
+
+@Singleton
+@Slf4j
+public class AdvancedGraphService {
```
[the remark this answers is not in the document]
Comment (loicmathieu) [reply core/src/main/java/io/kestra/core/services/AdvancedGraphService.java:18]: Well, we can still merge the two or name this one differently as this is really a smell to have two graph service, one being a static utility and the other one an injected bean.

File: core/src/main/java/io/kestra/core/models/tasks/Task.java [hunk 2]
```diff
@@ -30,6 +31,7 @@
 @Introspected
 @JsonInclude(JsonInclude.Include.NON_DEFAULT)
 abstract public class Task {
+    @Setter
```
[the remark this answers is not in the document]
Comment (Skraye) [reply core/src/main/java/io/kestra/core/models/tasks/Task.java:34]: Maybe juste create a new task with the good idea and replace it ?

**Comment by tchiotludo** [comment]
@Skraye brian will refactor a lot this one, the current implementation break the immutability principle of Kestra and is to intrusive on the Flowable task.

File: core/src/main/java/io/kestra/core/models/hierarchies/GraphCluster.java [hunk 3]
```diff
@@ -16,20 +17,15 @@ public class GraphCluster extends AbstractGraph {
     protected RelationType relationType;
 
     @JsonIgnore
-    private final GraphClusterRoot root;
+    protected final GraphClusterRoot root;
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/models/hierarchies/GraphCluster.java:20]: Can't it be private? (same for the 3 others)

File: core/src/main/java/io/kestra/core/models/hierarchies/GraphCluster.java [hunk 4]
```diff
@@ -39,25 +35,70 @@ public GraphCluster(String uid) {
...
+    @Override
+    public String getUid() {
+        return "cluster_" + super.getUid();
+    }
+
+    @Override
+    public void setUid(String uid) {
+        graph.nodes().stream().filter(node -> (!(node instanceof GraphClusterRoot) && !(node instanceof GraphClusterEnd))
+                || node.equals(this.root) || node.equals(this.end))
+            .forEach(node -> node.setUid(uid + node.uid.substring(this.uid.length())));
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/models/hierarchies/GraphCluster.java:91]: Can you explain this line?

File: core/src/main/java/io/kestra/core/utils/GraphUtils.java [hunk 5]
```diff
@@ -0,0 +1,455 @@
...
+import io.kestra.core.models.triggers.AbstractTrigger;
+import io.kestra.core.tasks.flows.Dag;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.commons.lang3.tuple.Triple;
+
+import java.util.*;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class GraphUtils {
```
Comment (loicmathieu) [inline core/src/main/java/io/kestra/core/utils/GraphUtils.java:19]: I didn't check this class, did you change some methods from the previous GraphService that I may have a look of is it the exact same?

**Review by loicmathieu — CHANGES_REQUESTED** [review]

File: core/src/main/java/io/kestra/core/utils/GraphUtils.java (hunk 5 above)
Comment (brian-mulier-p) [reply core/src/main/java/io/kestra/core/utils/GraphUtils.java:19]: Yes a lot, mainly the fillGraph method :'(

File: core/src/main/java/io/kestra/core/models/hierarchies/GraphCluster.java (hunk 4 above)
Comment (brian-mulier-p) [reply core/src/main/java/io/kestra/core/models/hierarchies/GraphCluster.java:91]: I'll add a comment

File: core/src/main/java/io/kestra/core/models/hierarchies/AbstractGraph.java [hunk 6]
```diff
@@ -32,4 +32,8 @@ public AbstractGraph(String uid) {
     public String getLabel() {
         return this.getUid();
     }
+
+    public void setUid(String uid) {
+        this.uid = uid;
+    }
```
[the remark this answers is not in the document]
Comment (brian-mulier-p) [reply core/src/main/java/io/kestra/core/models/hierarchies/AbstractGraph.java:38]: Sure I forgot to remove it after doing the custom setter for clusters

**Review by Skraye — APPROVED** [review]
LGTM

**Pull request merged**